### PR TITLE
unit-test-jest

### PR DESCRIPTION
**Pull request by Sam-Castaneda13**

### CODE REVIEW

File: unit-tests-jest/numbers.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+export function toDollars(amount: number): string {
+  return `$${amount}.00`;
+}
+
+/**
+ * Returns a new array of numbers where every entry has been divided by the
+ * given divisor. Does not modify the original array.
+ */
+export function divideBy(numbers: number[], divisor: number): number[] {
+  for (let i = 0; i < numbers.length; i++) {
```
Comment (RobertGardner) [inline unit-tests-jest/numbers.ts:25]: This modifies the array, while the description says it returns a new array and does not modify the original array.

File: unit-tests-jest/numbers.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+ */
+export function evenNumbers(numbers: number[]): number[] {
+  return numbers.filter((n) => n % 2 === 0);
+}
+
+/**
+ * Returns a number formatted in dollars and cents.
+ */
+export function toDollars(amount: number): string {
+  return `$${amount}.00`;
```
Comment (RobertGardner) [inline unit-tests-jest/numbers.ts]: This doesn't work with fractional amounts.

File: unit-tests-jest/numbers.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+ * Modifies an object by multiplying the value of each key, but only if that
+ * value is a number. Key names are arbitrary.
+ */
+export function multiplyBy(
+  obj: Record<string, unknown>,
+  multiplier: number
+): Record<string, unknown> {
+  const result: Record<string, unknown> = {};
+  Object.entries(obj).forEach(([key, value]) => {
+    result[key] = Number(value) * multiplier;
```
Comment (RobertGardner) [inline unit-tests-jest/numbers.ts]: This fails if `value` is a string because it puts NaN in there. It should not modify non-numbers.

File: unit-tests-jest/number.test.ts [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    const results = evenNumbers(numbers);
+    expect(results).toEqual([4, 22]);
+  });
+});
+
+describe('toDollars', () => {
+  it('returns number formatted as dollars', () => {
+    const money = 100;
+    const results = toDollars(money);
+    expect(results).toEqual('$100.00');
```
Comment (RobertGardner) [inline unit-tests-jest/number.test.ts]: Test fractional values, also. Any other edge cases you can think of?

File: unit-tests-jest/number.test.ts [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+  });
+});
+
+describe('divideBy', () => {
+  it('returns the divided numbers', () => {
+    const numbers = [25, 45, 64, 100, 235];
+    const divisor = 5;
+    const results = divideBy(numbers, divisor);
+    expect(results).toEqual([5, 9, 12.8, 20, 47]);
```
Comment (RobertGardner) [inline unit-tests-jest/number.test.ts]: You might want to check that the array was not modified.

File: unit-tests-jest/number.test.ts [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+  });
```
Comment (RobertGardner) [inline unit-tests-jest/number.test.ts:25]: Any other edge cases you can think of?

File: unit-tests-jest/number.test.ts [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+      five: 'one-hundred',
+    };
+    const multiplier = 5;
+    const results = multiplyBy(numbers, multiplier);
+    expect(results).toEqual({
+      one: 100,
+      two: 200,
+      three: 300,
+      four: 400,
+      five: NaN,
```
Comment (RobertGardner) [inline unit-tests-jest/number.test.ts]: This is incorrect. It should have value `five: 'one-hundred'`.

**Review by RobertGardner — CHANGES_REQUESTED** [review]

File: unit-tests-jest/numbers.ts [hunk 8]
```diff
@@ -9,18 +9,23 @@ export function evenNumbers(numbers: number[]): number[] {
  * Returns a number formatted in dollars and cents.
  */
 export function toDollars(amount: number): string {
-  return `$${amount}.00`;
+  if (amount % 1 !== 0) {
```
Comment (RobertGardner) [inline unit-tests-jest/numbers.ts:12]: It works fine without this check. `amount.toFixed(2)` will return, say, `3.00` for integers.

File: unit-tests-jest/numbers.ts [hunk 9]
```diff
@@ -33,7 +38,11 @@ export function multiplyBy(
 ): Record<string, unknown> {
   const result: Record<string, unknown> = {};
   Object.entries(obj).forEach(([key, value]) => {
-    result[key] = Number(value) * multiplier;
+    if (typeof value === 'string') {
```
Comment (RobertGardner) [inline unit-tests-jest/numbers.ts:41]: You should actually check the other way around. Only do the multiplication if it's a number.

**Review by RobertGardner — APPROVED** [review]
A couple of comments, but otherwise OK.
